homeassistant/components/kitchen_sink/__init__.py

```diff
@@ -110,6 +119,23 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     # Notify backup listeners
     hass.async_create_task(_notify_backup_listeners(hass), eager_start=False)
 
+    # Subscribe to labs feature updates for kitchen_sink preview repair
+    @callback
+    def _async_labs_updated(event: Event[EventLabsUpdatedData]) -> None:
```

Comment:
I wonder if this API could be simplified a bit by creating a helper function to create this listener function.

I'm thinking of something similar to the one below.
```python
    entry.async_on_unload(
        preview_features.async_listen(
            domain="kitchen_sink", 
            preview_feature="special_repair",
            listener=lambda: _async_update_special_repair(hass)
        )
    )
```
This way, the integration author does not have to wonder whether Futures Lab uses the event bus and what the message format is.

This will slightly extend the existing public API we currently have.
```
async_is_preview_feature_enabled(hass, DOMAIN, "special_repair"):
```